lib/afc/streams.js

```diff
@@ -59,29 +54,28 @@ class AfcWritableFileStream extends stream.Writable {
     this._destroyed = false;
   }
 
-  async _write (chunk, encoding, cb) {
-    try {
-      await this._afcService.writeFile(this._fileHandle, chunk);
-      cb();
-    } catch (e) {
-      if (this._autoDestroy) {
-        this.destroy(e);
-      }
-      cb(e);
-    }
+  _write (chunk, encoding, cb) {
+    this._afcService.writeFile(this._fileHandle, chunk)
+      .then(() => cb())
+      .catch((e) => {
+        if (this._autoDestroy) {
+          this.destroy(e);
```

Comment:
Should this get the callback, too? `this.destroy(e, cb)`?

Reply:
nope, the public method only accepts one argument

Reply:
Ok. What happens in `_autoDestroy` is true, with respect to this function, then? The callback for `_write` is never called, so does it hang?

Reply:
I cannot say for sure, although the previous implementation was the same. I didn't change this logic

Reply:
I think you did, though, by adding the `else`.

Reply:
true, reverted it back